community/modules/compute/gke-node-pool/variables.tf

```diff
@@ -0,0 +1,124 @@
+/**
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+variable "project_id" {
+  description = "The project ID to host the cluster in."
+  type        = string
+}
+
+variable "cluster_id" {
+  description = "projects/{{project}}/locations/{{location}}/clusters/{{cluster}}"
+  type        = string
+}
+
+variable "name" {
+  description = "The name of the node pool. If left blank, will default to the machine type."
+  type        = string
+  default     = null
+}
+
+variable "machine_type" {
+  description = "The name of a Google Compute Engine machine type."
+  type        = string
+  default     = "c2-standard-60"
+}
+
+variable "image_type" {
+  description = "The default image type used by NAP once a new node pool is being created. Use either COS_CONTAINERD or UBUNTU_CONTAINERD."
+  type        = string
+  default     = "COS_CONTAINERD"
+}
+
+# TODO
+variable "total_min_nodes" {
+  description = "Total minimum number of nodes in the NodePool."
+  type        = number
+  default     = 0
+}
+
+variable "total_max_nodes" {
+  description = "Total maximum number of nodes in the NodePool."
+  type        = number
+  default     = 1000
+}
+
+variable "threads_per_core" {
+  description = <<-EOT
+  Sets the number of threads per physical core. By setting threads_per_core
+  to 2, Simultaneous Multithreading (SMT) is enabled extending the total number
+  of virtual cores. For example, a machine of type c2-standard-60 will have 60
+  virtual cores with threads_per_core equal to 2. With threads_per_core equal
+  to 1 (SMT turned off), only the 30 physical cores will be available on the VM.
+
+  The default value of \"0\" will turn off SMT for supported machine types, and
+  will fall back to GCE defaults for unsupported machine types (t2d, shared-core
+  instances, or instances with less than 2 vCPU).
```

Comment:
Is this a standard behavior? It seems it could be 1 or 2 and not do anything with unsupported machines. 
Isn't the default always 2? unless it is a family where this is unsupported?